pkg/resource/deploy/source_eval.go

```diff
@@ -1866,6 +1871,14 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 			logging.V(5).Infof("ResourceMonitor.RegisterResource operation canceled, name=%s", name)
 			return nil, rpcerror.New(codes.Unavailable, "resource monitor shut down while waiting on step's done channel")
 		}
+		if result != nil && result.Failed {
+			return &pulumirpc.RegisterResourceResponse{
+				Urn:                  string(result.State.URN),
+				Id:                   string(result.State.ID),
+				Object:               nil,
```

Comment:
So the SDK gets a response that the resource did register, but just has no outputs, and might have an empty ID?
That seems suspect, I don't trust SDKs are going to handle this well vis-a-vis working out unknown vs undefined vs error'd output states.